fontface.go

```diff
@@ -164,85 +164,59 @@ func (family *FontFamily) Face(size float64, col color.Color, style FontStyle, v
 	r, g, b, a := col.RGBA()
 	return FontFace{
 		family:     family,
-		font:       font,
+		Font:       font,
 		Size:       size,
 		Style:      style,
 		Variant:    variant,
 		Color:      color.RGBA{uint8(r >> 8), uint8(g >> 8), uint8(b >> 8), uint8(a >> 8)},
 		deco:       deco,
 		Scale:      scale,
 		Voffset:    voffset,
-		fauxItalic: fauxItalic,
-		fauxBold:   fauxBold * size * scale,
+		FauxItalic: fauxItalic,
+		FauxBold:   fauxBold * size * scale,
 	}
 }
 
 // FontFace defines a font face from a given font. It allows setting the font size, its color, faux styles and font decorations.
 type FontFace struct {
 	family *FontFamily
-	font   *Font
+	Font   *Font
 
 	Size    float64
 	Style   FontStyle
 	Variant FontVariant
 	Color   color.RGBA
 	deco    []FontDecorator
 
-	Scale, Voffset, fauxBold, fauxItalic float64 // consequences of font style and variant
+	Scale, Voffset, FauxBold, FauxItalic float64 // consequences of font style and variant
 }
 
 // Equals returns true when two font face are equal. In particular this allows two adjacent text spans that use the same decoration to allow the decoration to span both elements instead of two separately.
 func (ff FontFace) Equals(other FontFace) bool {
-	return ff.font == other.font && ff.Size == other.Size && ff.Style == other.Style && ff.Variant == other.Variant && ff.Color == other.Color && reflect.DeepEqual(ff.deco, other.deco)
+	return ff.Font == other.Font && ff.Size == other.Size && ff.Style == other.Style && ff.Variant == other.Variant && ff.Color == other.Color && reflect.DeepEqual(ff.deco, other.deco)
 }
 
 // Name returns the name of the underlying font
 func (ff FontFace) Name() string {
-	return ff.font.name
-}
-
-// FontMetrics contains a number of metrics that define a font face.
-type FontMetrics struct {
-	Size       float64
-	LineHeight float64
-	Ascent     float64
-	Descent    float64
-	XHeight    float64
-	CapHeight  float64
+	return ff.Font.name
 }
 
 // Metrics returns the font metrics. See https://developer.apple.com/library/archive/documentation/TextFonts/Conceptual/CocoaTextArchitecture/Art/glyph_metrics_2x.png for an explanation of the different metrics.
 func (ff FontFace) Metrics() FontMetrics {
-	buffer := &sfnt.Buffer{}
-	m, _ := ff.font.sfnt.Metrics(buffer, toI26_6(ff.Size*ff.Scale), font.HintingNone)
+	m := ff.Font.Metrics(ff.Size * ff.Scale)
 	return FontMetrics{
-		Size:       ff.Size,
-		LineHeight: math.Abs(fromI26_6(m.Height)),
-		Ascent:     math.Abs(fromI26_6(m.Ascent)),
-		Descent:    math.Abs(fromI26_6(m.Descent)),
-		XHeight:    math.Abs(fromI26_6(m.XHeight)),
-		CapHeight:  math.Abs(fromI26_6(m.CapHeight)),
+		LineHeight: math.Abs(m.LineHeight),
+		Ascent:     math.Abs(m.Ascent),
```

Comment:
I am not sure why this is using `math.Abs`, so I let it there.

Reply:
Because these values can be negative for some fonts (as in, from the baseline upwards is along a negative axis, so ascent should be negative), but they should always mean a distance which is positive of course.